src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let copyState = { ...state };
+  const result = [];
+
+  actions.forEach((elem) => {
+    const { type, extraData, keysToRemove } = elem;
+
+    switch (type) {
+      case 'addProperties':
+        copyState = Object.assign(copyState, extraData);
```

Comment:
Using `Object.assign(copyState, extraData)` directly modifies `copyState`. Since `copyState` is already a clone of `state`, this is fine, but be aware that `Object.assign` modifies the target object. If immutability is a concern, consider using `copyState = { ...copyState, ...extraData }`.